### ISSUE TITLE
Add support for pnpm in flatpak-node-generator

### ISSUE THREAD

**Issue by RushingAlien** [flatpak/flatpak-builder-tools#383]
### flatpak-builder version

1.2.3

### Linux distribution and version

Arch, rolling

### Affected flatpak-builder tool

node/flatpak-node-generator.py

### Source repository URL

https://github.com/Vencord/Vesktop

### Flatpak-builder manifest URL

https://github.com/RushingAlien/vesktop-flatpak/tree/build-from-source

### Description

Some node projects now use pnpm instead of npm, I've tried to workaround it with a few methods, but to no avail. Pnpm's package lock is a yaml file called `pnpm-lock.json`

A workaround i've tried involves generating using npm to generate package-lock.json then use flatpak-node-generator npm tool then try to install dependencies with either npm or pnpm.

**Comment by ImUrX** [comment]
a lot of popular projects do use it qwq

**Comment by ndom91** [comment]
Been running into this and the existing solutions didn't work for me, so I've been working on a `pnpm-lock.yaml` to `package-lock.json` converter which supports pnpm workspaces and catalog dependencies here: https://github.com/ndom91/pnpm-lock-to-package-lock. 

Would love any input / testing from folks if you've got a use for it 😊

**Comment by nukeop** [comment]
I'm going to need this - my project is a Tauri, pnpm-only application. My only choice right now is repackaging a binary, I can't create a flatpak from source. And given how much better pnpm is, more and more projects are going to be blocked by this.

I did some initial research on supporting pnpm and it doesn't seem too hard. I might take a shot soon.

**Comment by nukeop** [comment]
NN708's concern is outdated. The linked pnpm issue from 2020 (https://github.com/pnpm/pnpm/issues/2978) is no longer relevant.

pnpm's offline mode (--frozen-lockfile --offline) works perfectly with a pre-populated content-addressable store. The store can be reconstructed from tarballs using straightforward file operations. Extract each tarball, SHA-512 hash each file, write to `{store}/v10/files/{hash[:2]}/{hash[2:]}`, and create an index JSON. This is conceptually identical to what the npm provider already does with _cacache.

I have a working implementation in progress.
